src/h.ts

```diff
@@ -34,7 +34,7 @@ export function h(sel: any, b?: any, c?: any): VNode {
   }
   if (is.array(children)) {
     for (i = 0; i < children.length; ++i) {
-      if (is.primitive(children[i])) children[i] = vnode(undefined, undefined, undefined, children[i], undefined);
```

Comment:
creating a new object everytime while we were using `undefined` before will be a perf penalty.

snabbdom never mutates the VNodeData structure itself, in which case reusing an empty object should be ok (and favored), but this object might need to be marked as non-extensible so 3rd party modules might not attempt to mutate it, this itself might be problematic (non-backward compatible).

Reply:
@caridy making `data` "non-extensible" (immutable?) would not only be a breaking change, it would be undesirable.
Users might want to mutate the `data` of VNodes returned from `h`. I know because I do it.

Oh, but this line is only for text VNodes.
I guess there is no foreseeable reason to mutate a text VNode's `data`. Is there?

So, a top scope empty object that will be reused as `data` of all text VNodes?

Should I `Object.freeze` it?

As far as TypeScript, I don't see a reasonable way to type the `data` of a text VNode as an empty object. The only way I can see is that we have a new interface, `TextVNode` and that seems kind of overkill to me. So how about just document this caveat and `Object.freeze` to prevent mistakes?

Is there a standard way of running benchmarks, so I could test the possible perf penalty of this new object, at all? Who knows—perhaps engine optimizations make this a non-issue.